signer-service/src/api/services/sep10.service.js

```diff
@@ -0,0 +1,95 @@
+const { Keypair } = require('stellar-sdk');
+const { TransactionBuilder, Networks } = require('stellar-sdk');
+const { fetchTomlValues } = require('../helpers/anchors');
+
+const { TOKEN_CONFIG } = require('../../constants/tokenConfig');
+const { CLIENT_SECRET } = require('../../constants/constants');
+
+const NETWORK_PASSPHRASE = Networks.PUBLIC;
+
+exports.signSep10Challenge = async (challengeXDR, outToken, clientPublicKey) => {
+  const clientDomainStellarKeypair = Keypair.fromSecret(CLIENT_SECRET);
+
+  const { signingKey: anchorSigningKey } = await fetchTomlValues(TOKEN_CONFIG[outToken].tomlFileUrl);
+
+  const transactionSigned = new TransactionBuilder.fromXDR(challengeXDR, NETWORK_PASSPHRASE);
+  if (transactionSigned.source !== anchorSigningKey) {
+    throw new Error(`Invalid source account: ${transactionSigned.source}`);
+  }
+  if (transactionSigned.sequence !== '0') {
+    throw new Error(`Invalid sequence number: ${transactionSigned.sequence}`);
+  }
+
+  // See https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#success
+  // memo field should be empty as we assume (in this implementation) that we use the ephemeral
+  // to authenticate. But no memo sub account derivation.
+  if (transactionSigned.memo.value === '') {
+    throw new Error('Memo does not match');
+  }
```

Comment:
I think the condition needs to be flipped and we can give a better error message.

```suggestion
  if (transactionSigned.memo.value !== '') {
    throw new Error('Memo should be empty');
  }
```

Reply:
I will change the message, but we actually expect it to be empty. It's very important because if the UI were to send the memo they want, we would essentially validate their identity, without any check they provide us.

On the memo PR we will validate it.

Reply:
Okay but if you want the memo to be empty then we need to flip the condition because otherwise you will throw an error here if it's empty, no?

Reply:
Actually you're right 🤦. I had the condition the other way around. And the reason it was working is because it's not empty but `null`. So I corrected to _expect_ a null.